### PR TITLE
Refactor queue

### PR DESCRIPTION
**Pull request by wigging**
This pull request removes the NATS queue and associated code since it is not used in zambeze. The abstract queue and associated queue factory are also removed. The entire queue directory is removed. The queue module for RabbitMQ is now located at the top-level of the orchestration directory.

### CODE REVIEW

**Comment by wigging** [comment]
I removed the `on push` for the workflows because the tests and checks were being run twice: once for push and once for pull request.

**Comment by wigging** [comment]
More changes... I removed the utils and auth directories. Renamed `globus_auth.py` to just `auth.py`.

File: examples/imagemagick_globus.py [hunk 1]
```diff
@@ -50,10 +47,13 @@ def main():
         override_existing=False,
     )
 
-    campaign.add_activity(activity)
-    campaign.add_activity(transfer)
+    # create campaign and dispatch it
+    campaign = Campaign(
```
Comment (tskluzac) [inline examples/imagemagick_globus.py:51]: I think switching from adding activities as its own function  to part of the Campaign instantiation (as you did) is cleaner. And at this point I believe this is okay functionally, but soon we will have to have dynamically-added tasks that might require the campaign having more activities added as the workflow is executed. So we may have to take slightly more care to ensure that the activity object in the Campaign is mutable.

**Review by tskluzac — APPROVED** [review]
Looks good! I think these changes greatly simplify the code base, both internally and for the users.

File: examples/imagemagick_globus.py (hunk 1 above)
Comment (wigging) [reply examples/imagemagick_globus.py:51]: The `Campaign` class has an `add_activity` method to add more activities. So this should be able to handle dynamic tasks but we'll need to develop some examples and tests to see what the limitations are.

**Pull request merged**